hail/src/main/scala/is/hail/rvd/RVD.scala

```diff
@@ -469,7 +493,29 @@ class RVD(
     if (n == 0)
       return RVD.empty(sparkContext, typ)
 
-    val newRDD = crdd.tail(n, partitionCounts)
+    val (idxFirst, nDrop) = partitionCounts match {
+      case Some(pcs) =>
+        getPCSubsetOffset(n, pcs.reverseIterator) match {
+          case Some(PCSubsetOffset(idx, _, nDrop)) => (pcs.length - idx - 1) -> nDrop
+          case None => return this
+        }
+      case None =>
+        val crddBoundary = crdd.boundary
+        val PCSubsetOffset(idx, _, nDrop) =
+          incrementalPCSubsetOffset(n, Range.inclusive(getNumPartitions - 1, 0, -1))(
+            crddBoundary.runJob(getIteratorSize, _)
+          )
+        idx -> nDrop
+    }
+
+    val newRDD = crdd.mapPartitionsWithIndex({ case (i, it) =>
+      if (i == idxFirst)
+        it.drop(nDrop.toInt)
```

Comment:
I *believe* `it.drop` is going to have issues with region clearing, since by default it's just going to call `next()` and `hasNext()` on the iterator.

Reply:
ack, good point.